velox/exec/benchmarks/OrderByBenchmarkUtil.cpp

```diff
@@ -0,0 +1,308 @@
+        int iterations,
+        int numKeys)> benchmark) {
+  std::vector<RowTypePtr> rowTypes = smallintRowTypes(noPayload);
+  std::vector<int> numKeys = {1, 2, 3, 4};
+  addBenchmark(
+      noPayload ? "no-payload" : "payload",
+      "smallint",
+      batchSizes,
+      rowTypes,
+      numKeys,
+      numIterations,
+      benchmark);
+}
+
+void smallSmallint(std::function<void(
+                       const std::string& testName,
+                       size_t numRows,
+                       const RowTypePtr& rowType,
+                       int iterations,
+                       int numKeys)> benchmark) {
+  // For small dateset, iterations need to be large enough to ensure that the
+  // benchmark runs for enough time.
+  const auto iterations = 100'000;
+  const std::vector<vector_size_t> batchSizes = {10, 50, 100, 500};
+  smallint(true, iterations, batchSizes, benchmark);
+}
+
+void smallSmallintWithPayload(std::function<void(
+                                  const std::string& testName,
+                                  size_t numRows,
+                                  const RowTypePtr& rowType,
+                                  int iterations,
+                                  int numKeys)> benchmark) {
+  const auto iterations = 100'000;
+  const std::vector<vector_size_t> batchSizes = {10, 50, 100, 500};
+  smallint(false, iterations, batchSizes, benchmark);
+}
+
+void largeSmallint(std::function<void(
+                       const std::string& testName,
+                       size_t numRows,
+                       const RowTypePtr& rowType,
+                       int iterations,
+                       int numKeys)> benchmark) {
+  const auto iterations = 10;
+  const std::vector<vector_size_t> batchSizes = {
+      1'000, 10'000, 100'000, 1'000'000};
+  smallint(true, iterations, batchSizes, benchmark);
+}
+
+void largeSmallintWithPayloads(std::function<void(
+                                   const std::string& testName,
+                                   size_t numRows,
+                                   const RowTypePtr& rowType,
+                                   int iterations,
+                                   int numKeys)> benchmark) {
+  const auto iterations = 10;
+  const std::vector<vector_size_t> batchSizes = {
+      1'000, 10'000, 100'000, 1'000'000};
+  smallint(false, iterations, batchSizes, benchmark);
+}
+} // namespace
+
+RowVectorPtr OrderByBenchmarkUtil::fuzzRows(
+    const RowTypePtr& rowType,
+    vector_size_t numRows,
+    int numKeys,
+    memory::MemoryPool* pool) {
+  VectorFuzzer fuzzer({.vectorSize = static_cast<size_t>(numRows)}, pool);
+  VectorFuzzer fuzzerWithNulls(
+      {.vectorSize = static_cast<size_t>(numRows),
+       .nullRatio = FLAGS_data_null_ratio},
+      pool);
+  std::vector<VectorPtr> children;
+
+  // Fuzz keys: for front keys (column 0 to numKeys -2) use high
+  // nullRatio to enforce all columns to be compared.
+  for (auto i = 0; i < numKeys - 1; ++i) {
```

Comment:
Does this reflect the actual use scenarios? Can we simplify the case with null and not-null to ease analysis? Thanks!

Reply:
Actual cases columns maybe null or not null, so we need both of them. The default null ratio 0.7 is not actual use scenarios, it is just used to compute all the columns in sort, we can change it by FLAGS_data_null_ratio

Reply:
Then why not have all the key columns with null ratio set to keep it simple?